### DISCUSSION TITLE
Multi-window react application?

### DISCUSSION THREAD

**Discussion by erikpa1** [tauri-apps/tauri#5404] [Q&A]
Hi,

I know tauri has support for multi window, but thing is I use React, which is based on single html file, do you have some experiences with it? What are possible options you have experienced? My guesses was:

- create multiple react app and link only build to tauri (not very effective I think) ?
- loading index html, with some params like process arguments to dispatch view in main react window ?

**Comment by xuchaoqian** [comment]
If you use webpack, you can set multi-entries for it:

![image](https://user-images.githubusercontent.com/1064052/196004672-ade41305-6ed4-4c8f-9dbd-1af6be679cd0.png)

and then inject chunks into html:

![image](https://user-images.githubusercontent.com/1064052/196004757-b399a3e7-a93a-4294-a090-f9ff2ff6a0d9.png)

**Reply by erikpa1** [reply]
Nah, it requires eject my react app which is the worst thing in beggining of  development you can make.

**Reply by FabianLars** [reply]
If you don't want to eject your app or use a different tooling like vitejs then the only thing I can think of is using react-router and a different route per window.

**Reply by erikpa1** [reply]
@FabianLars I think it will work, but how react app will know path ? 

Some additional parameter here?

```

    let local_window = tauri::WindowBuilder::new(
      app,
      "local",
      tauri::WindowUrl::App("index.html".into())
```

like `index.html/#/mypath` ?

Using same app with different route is also interesting for me,  because I want also web app version :)

**Reply by FabianLars** [reply]
Yeah exactly! `index.html/#/Route` is what I'm using too (with react router in hash mode)

**Reply by erikpa1** [reply]
Ok thanks, I will try, thanks for your time, have a nice day :)

**Reply by MohamedMujtaba** [reply]
can you provide the code for how did you do it ?

**Reply by erikpa1** [reply]
@MohamedMujtaba  https://github.com/erikpa1/TurtleEngineFrontend/blob/main/src-tauri/src/main.rs  maybe if it will help
